### PR TITLE
QR with column pivoting solver

### PR DESCRIPTION
**Pull request by kyungjoo-kim**
In the previous PR, I also provide a UTV solver for rank deficient problems. Previous unit test only covers a single right hand side vector. This PR include multiple right hand side case.

### CODE REVIEW

**Comment by kyungjoo-kim** [comment]
kokkos dev spot check
```
#######################################################
PASSED TESTS
#######################################################
clang-4.0.1-Pthread_Serial-release build_time=319 run_time=126
clang-7.0.1-Cuda_OpenMP-release build_time=1077 run_time=263
cuda-9.2-Cuda_OpenMP-release build_time=1137 run_time=286
gcc-5.3.0-OpenMP-release build_time=252 run_time=81
gcc-7.3.0-Serial-release build_time=201 run_time=65
intel-16.0.3-Pthread-release build_time=306 run_time=66
intel-16.0.3-Serial-release build_time=308 run_time=76
intel-17.0.1-OpenMP-release build_time=430 run_time=53
```

File: unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+#include "KokkosKernels_TestUtils.hpp"
+
+using namespace KokkosBatched;
+
+namespace Test {
+
+  template<typename DeviceType,
+           typename MatrixViewType,
+           typename VectorViewType,
+	   typename PivViewType,
```
Comment (ndellingwood) [inline unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp:23]: Minor format comment, this line is unaligned; not necessary to change in the PR, good to cleanup later.

File: unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+           typename WorkViewType,
+           typename AlgoTagType>
+  void impl_test_batched_solve_utv2(const int N, const int BlkSize) {
+    typedef typename MatrixViewType::non_const_value_type value_type;
+    typedef Kokkos::Details::ArithTraits<value_type> ats;
+    //const value_type one(1);
```
Comment (ndellingwood) [inline unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp:144]: Minor format comment, this line can be removed; not necessary to change in the PR, just a note to cleanup later.

File: unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+	for (int i=0;i<BlkSize;++i) {
+	  value_type tmp(0);
+	  for (int j=0;j<BlkSize;++j) {	
+	    tmp += a_host(k,j,i)*w_host(k,j);
+	  }
+	  residual += ats::abs(tmp)*ats::abs(tmp);
+	  norm += ats::abs(b_host(k,i,l))*ats::abs(b_host(k,i,l));	
+	}
+      }
+      //printf("norm %e, residual %e, rel res %e\n", norm, residual, residual/norm);
```
Comment (ndellingwood) [inline unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp:206]: Minor format comment, this line can be removed; not necessary to change in the PR, just a note to cleanup later.

File: unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+#if defined(KOKKOSKERNELS_INST_LAYOUTLEFT)
+  {
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutLeft, DeviceType> MatrixViewType;
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutLeft, DeviceType> VectorViewType;
+    typedef Kokkos::View<IntType**,   Kokkos::LayoutLeft, DeviceType> PivViewType;
+    typedef Kokkos::View<ValueType**, Kokkos::LayoutRight,DeviceType> WorkViewType;
+    Test::impl_test_batched_solve_utv2<DeviceType,MatrixViewType,VectorViewType,
+				       PivViewType,WorkViewType,AlgoTagType>(     0, 10);
+    for (int i=1;i<10;++i) {                                                                                        
+      //printf("Testing: LayoutLeft,  Blksize %d\n", i);
```
Comment (ndellingwood) [inline unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp:227]: Minor format comment, this line can be removed; not necessary to change in the PR, just a note to cleanup later.

File: unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+#if defined(KOKKOSKERNELS_INST_LAYOUTRIGHT)
+  {
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutRight,DeviceType> MatrixViewType;
+    typedef Kokkos::View<ValueType***,Kokkos::LayoutRight,DeviceType> VectorViewType;
+    typedef Kokkos::View<IntType**,   Kokkos::LayoutRight,DeviceType> PivViewType;
+    typedef Kokkos::View<ValueType**, Kokkos::LayoutRight,DeviceType> WorkViewType;
+    Test::impl_test_batched_solve_utv2<DeviceType,MatrixViewType,VectorViewType,
+				       PivViewType,WorkViewType,AlgoTagType>(     0, 10);
+    for (int i=1;i<10;++i) {                                                                                        
+      //printf("Testing: LayoutRight, Blksize %d\n", i);
```
Comment (ndellingwood) [inline unit_test/batched/Test_Batched_TeamVectorSolveUTV2.hpp:242]: Minor format comment, this line can be removed; not necessary to change in the PR, just a note to cleanup later.

**Comment by kyungjoo-kim** [comment]
@ndellingwood Thank you.  I just leave the print statement comment for a convenience to narrow down the problem. If this does not hurt much program readability, I would like to leave the comment.

**Pull request merged**